bioptim/optimization/optimal_control_program.py

```diff
@@ -380,7 +380,17 @@ def __init__(
         # Prepare the variable mappings
         if variable_mappings is None:
             variable_mappings = BiMappingList()
-        NLP.add(self, "variable_mappings", variable_mappings, True)
+
+        variable_mappings_per_phase = [{} for _ in range(self.n_phases)]
+        for mappings in variable_mappings:
+            for mapping in mappings:
```

Comment:
`mapping` seems to be a `key`
if so please use that variable name since it is a bit confusing to see: `mappings[mapping]`